crates/forge/src/fuzzer.rs

```diff
@@ -0,0 +1,171 @@
+    fuzzer_run_params: FuzzerRunParams,
+}
+
+impl RandomFuzzer {
+    pub fn new(
+        seed: u64,
+        runs_number: u32,
+        args_number: usize,
+        low: BigUint,
+        high: BigUint,
+    ) -> Self {
+        let rng = StdRng::seed_from_u64(seed);
+
+        let mut fuzzer = RandomFuzzer {
+            rng,
+            fuzzer_run_params: FuzzerRunParams::default(),
+        };
+        fuzzer.set_fuzzer_run_params(runs_number, args_number, low, high);
+
+        fuzzer
+    }
+
+    fn set_fuzzer_run_params(
+        &mut self,
+        runs_number: u32,
+        args_number: usize,
+        low: BigUint,
+        high: BigUint,
+    ) {
+        assert!(low < high);
+        assert!(runs_number >= 3);
+
+        self.fuzzer_run_params.runs_executed = 0;
+        self.fuzzer_run_params.low = low;
+        self.fuzzer_run_params.high = high;
+        self.fuzzer_run_params.runs_number = runs_number;
+        self.fuzzer_run_params.args_number = args_number;
+
+        let runs_with_min_value: Vec<u32> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .map(|_| self.rng.gen_range(1..=self.fuzzer_run_params.runs_number))
+            .collect();
+        let runs_with_max_value: Vec<u32> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .zip(&runs_with_min_value)
+            .map(|(_, &run_with_min)| {
+                let run_with_max = self.rng.gen_range(1..=self.fuzzer_run_params.runs_number);
+                if run_with_max == run_with_min {
+                    run_with_min + 1 % runs_number
+                } else {
+                    run_with_max
+                }
+            })
+            .collect();
+
+        self.fuzzer_run_params.runs_with_max_value = runs_with_max_value;
+        self.fuzzer_run_params.runs_with_min_value = runs_with_min_value;
+    }
+
+    pub fn next_felt252_args(&mut self) -> Vec<Felt252> {
+        assert!(self.fuzzer_run_params.runs_executed < self.fuzzer_run_params.runs_number);
+
+        self.fuzzer_run_params.runs_executed += 1;
+        let current_run = self.fuzzer_run_params.runs_executed;
+
+        let args_values: Vec<Felt252> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .enumerate()
+            .map(|(i, _)| {
+                Felt252::from(
+                    if self.fuzzer_run_params.runs_with_min_value[i] == current_run {
+                        self.fuzzer_run_params.low.clone()
+                    } else if self.fuzzer_run_params.runs_with_max_value[i] == current_run {
+                        self.fuzzer_run_params.high.clone() - BigUint::one()
+                    } else {
+                        self.rng.gen_biguint_range(
+                            &self.fuzzer_run_params.low,
+                            &self.fuzzer_run_params.high,
+                        )
+                    },
+                )
+            })
+            .collect();
+
```

Comment:
We're constructing an iterator every time someone calls `next_felt252_args` + this code is quite confusing. Isn't there a way to handle these in without enumerating and mapping (just a loop maybe or some other approach)?

Reply:
Creating an iterator doesn't hurt I think, but I can try to make the code less confusing